libs/core/src/cart/cart-item.ts

```diff
@@ -1,5 +1,8 @@
+import { ProductImage } from "../product/product-image";
+
 export interface CartItem {
   item_id: number;
+  image: ProductImage;
```

Comment:
I think I'd need to make a getProduct service call, right? In case you navigate directly to the cart page?

Reply:
I've been thinking about this. I think I'd need to have a selector for the image, then if the product isn't in state I'd need to make a service call for the product, while the cart waits on all this. Is that how it's normally done? I'm gonna just start doing that

Reply:
Been going through this, and I think we need to somehow have cart state hold the first image for every product in the cart. Because of the edge-case of "if the user navigates to a cart with items directly", we would need to fire off an action from cart effects that tried to select the image from a product in state. That selection action would fail and we'd then trigger the GetProduct call. But we now have a dependency of CartState on ProductState, which I think we want to avoid. I'm not sure how big of a problem matching up with magento and shopify models is though